tests/kubeutils.py

```diff
@@ -0,0 +1,117 @@
+import time
+
+from kubernetes import client
+from kubernetes import watch
+from kubernetes.client import rest
+from kubernetes.stream import stream
+
+
+RESOURCE_READY = 0
+RESOURCE_EMPTY = 1
+RESOURCE_MISSING = 2
+
+
+def assert_resource_availability(resource, node, status, timeout=0):
+    v1 = client.CoreV1Api()
+
+    w = watch.Watch()
+    for event in w.stream(v1.list_node, timeout_seconds=timeout):
```

Comment:
How do you know if the event has not been already dispatched before calling this?

Reply:
It first lists all existing as "ADDED".